cl/search/tests/tests_es_recap.py

```diff
@@ -3090,10 +3176,153 @@ def test_date_filed_sorting_function_score(self) -> None:
             docket_old.delete()
             docket_null_date_filed.delete()
 
+    @override_settings(SEARCH_API_PAGE_SIZE=1)
+    @mock.patch(
+        "cl.search.api_views.DocketESResultSerializer",
+        new=DocketESResultSerializerTest,
+    )
+    @mock.patch(
+        "cl.search.api_views.BaseRECAPDocumentESResultSerializer",
+        new=BaseRECAPDocumentESResultSerializerTest,
+    )
+    @mock.patch(
+        "cl.search.api_views.RECAPESResultSerializer",
+        new=RECAPESResultSerializerTest,
+    )
+    @mock.patch(
+        "cl.search.api_utils.set_results_highlights",
+        side_effect=mock_set_results_highlights,
+    )
+    def test_stable_scores_date_sort(
+        self, mock_set_results_highlights
+    ) -> None:
+        """Test if the function's score remains stable when used for sorting by
+        dateFiled asc and entry_date_filed asc across pagination, avoiding
+        the impact of time running on the scores.
+        """
+
+        # Query string, order by dateFiled desc
+        search_params = {
+            "q": "SUBPOENAS SERVED",
+            "order_by": "dateFiled asc",
+            "highlight": False,
+        }
+
+        # string query date_filed asc
+        date_filed_asc_string_query = search_params
+        # match_all query date_filed asc
+        date_filed_asc_match_all = search_params.copy()
+        del date_filed_asc_match_all["q"]
+        # string query entry_date_filed asc
+        entry_filed_asc_string_query = search_params.copy()
+        entry_filed_asc_string_query["order_by"] = "entry_date_filed asc"
+        # match_all query entry_date_filed asc
+        entry_filed_asc_match_all = entry_filed_asc_string_query.copy()
+        del entry_filed_asc_match_all["q"]
+
+        tests_params = [
+            date_filed_asc_string_query,
+            date_filed_asc_match_all,
+            entry_filed_asc_string_query,
+            entry_filed_asc_match_all,
+        ]
+        # Generate tests for all the search types.
+        all_tests = []
+        for test_param in tests_params:
+            for search_type in [
+                SEARCH_TYPES.RECAP,
+                SEARCH_TYPES.DOCKETS,
+                SEARCH_TYPES.RECAP_DOCUMENT,
+            ]:
+                test_param_copy = test_param.copy()
+                test_param_copy["type"] = search_type
+                all_tests.append(test_param_copy)
+
+        for search_params in all_tests:
+            with self.subTest(
+                search_params=search_params, msg="Test stable scores."
+            ):
+                original_datetime = now().replace(day=1, hour=5, minute=0)
+                # Two first-page requests (no cursor) made on the same day will now
+                # have consistent scores because scores are now computed based on the
+                # same day's date.
+                with time_machine.travel(original_datetime, tick=False):
+                    r = self.client.get(
```

Comment:
We can potentially improve code reusability by implementing the [shift](https://github.com/adamchainz/time-machine?tab=readme-ov-file#shiftdelta) helper function. This function encapsulates the logic to move the current time by a given offset. By using this function before triggering an API request, we can avoid repeating the "time_machine" contexts throughout the code.

We would need to apply the following refactor to the code:

```python
original_datetime = now().replace(day=1, hour=5, minute=0)

for search_params in all_tests:
   with self.subTest(
       search_params=search_params, msg="Test stable scores."
   ), time_machine.travel(original_datetime, tick=False) as traveler:
   
        # first API request 
        date_score_1 = r.data["results"][0]["date_score"]


       traveler.shift(datetime.timedelta(minutes=1))
       # 2nd API request
       date_score_2 = r.data["results"][0]["date_score"]

       ...
```  

what do you think?

Reply:
Yeah I agree, this is a perfect use case for `shift`.